File: questModule/src/main/java/ru/nekit/android/qls/lockScreen/content/AbstractLockScreenContentMediator.java
package ru.nekit.android.qls.lockScreen.content;

import android.support.annotation.NonNull;

import ru.nekit.android.qls.lockScreen.TransitionChoreograph;

public abstract class AbstractLockScreenContentMediator {

    TransitionChoreograph mTransitionChoreograph;

    public void setTransitionChoreograph(TransitionChoreograph choreograph) {
        mTransitionChoreograph = choreograph;
    }

    @NonNull
    public abstract ILockScreenContentContainerViewHolder getViewHolder();

    public abstract void destroy();

    public abstract void detachView();

    public abstract void attachView();

}

File: questPresentation/src/main/java/ru/nekit/android/qls/lockScreen/mediator/common/AbstractLockScreenContentMediator.kt
package ru.nekit.android.qls.lockScreen.mediator.common

import io.reactivex.disposables.CompositeDisposable
import ru.nekit.android.qls.quest.providers.IQuestContextSupport

abstract class AbstractLockScreenContentMediator : IQuestContextSupport {

    override var disposableMap: MutableMap<String, CompositeDisposable> = HashMap()

    abstract val viewHolder: ILockScreenContentViewHolder

    abstract fun deactivate()

    abstract fun detachView()

    abstract fun attachView(callback: () -> Unit)

}